apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient3-plugin/src/main/java/co/elastic/apm/agent/httpclient/v3/HttpClient3Instrumentation.java

```
@@ -0,0 +1,138 @@
+import org.apache.commons.httpclient.URI;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+/**
+ * Instruments {@link org.apache.commons.httpclient.HttpMethodDirector#executeMethod(HttpMethod)}
+ */
+@SuppressWarnings("JavadocReference") // instrumented class is package-private
+public class HttpClient3Instrumentation extends TracerAwareInstrumentation {
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Arrays.asList("http-client", "apache-httpclient");
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.apache.commons.httpclient.HttpMethodDirector");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("executeMethod");
+    }
+
+    @Override
+    public String getAdviceClassName() {
+        return "co.elastic.apm.agent.httpclient.v3.HttpClient3Instrumentation$HttpClient3Advice";
+    }
+
+    public static class HttpClient3Advice {
+
+        @Nullable
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static Object onEnter(@Advice.Argument(0) HttpMethod httpMethod) {
+            final AbstractSpan<?> parent = TracerAwareInstrumentation.tracer.getActive();
+            if (parent == null) {
+                return null;
+            }
+
+            String host;
+            String uri;
+            String protocol;
+            int port;
+            try {
+                URI httpClientURI = httpMethod.getURI();
+                host = httpClientURI.getHost();
+                uri = httpClientURI.toString();
+                protocol = httpClientURI.getScheme();
+                port = httpClientURI.getPort();
+            } catch (Exception e) {
+                // silently ignored
```

Why is this silently ignored? Is there anything expected to be thrown here?